src/main.rs

```diff
@@ -0,0 +1,134 @@
+mod args;
+mod config;
+mod ignore;
+mod remote_command;
+mod sync;
+mod time;
+
+use args::Args;
+use config::Config;
+use ignore::*;
+use remote_command::execute_remote_command as execute_remote_command_impl;
+use std::env;
+use std::path::PathBuf;
+use std::process;
+use std::time::Instant;
+use sync::*;
+use time::*;
+
+fn main() {
+    println!(":: Mainframer v2.1.0\n");
```

Comment:
The version is duplicated here is there some way to pass in values into rust from tooling. e.g. custom defines in c++ or BuildConfig from Java.

Reply:
Yeah, cargo publishes package version thru build-time environment variables. 
https://doc.rust-lang.org/cargo/reference/environment-variables.html

Reply:
Yep yep, in plans in a separate PR, thanks for link!